tests/utils.py

```diff
@@ -173,6 +173,7 @@ def __init__(self, source=None, source_regex=None, prefix=''):
             assert source_regex is None
         self.line_pattern = re.compile(
             r"""^%s(?P<indent>(?: {4})*)[0-9:.]{15} """
+            r"""(?P<threads_info>[0-9]*-[A-Za-z_]* )?"""
```

Comment:
1. Should be `+`, not `*`. Should probably be the same for the rest of the regex as well, don't know what the reasoning was.
2. Need to allow for more than one space.
3. Need to allow for `-` and numbers in the thread name. For me, `threading.Thread().name` gives `'Thread-357'`.

You also need to write a test that has multiple threads with names of different lengths.

Reply:
Agreed, except I'm not sure about 3. But it's a small point anyway, so I don't mind about that.